modules/TEP.py

```diff
@@ -1,76 +1,74 @@
+    # Initialize spark session
+    spark_mem_limit = detect_spark_memory_limit()
+    spark_conf = (
+        SparkConf()
+        .set('spark.driver.memory', f'{spark_mem_limit}g')
+        .set('spark.executor.memory', f'{spark_mem_limit}g')
+        .set('spark.driver.maxResultSize', '0')
+        .set('spark.debug.maxToStringFields', '2000')
+        .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+    )
+    spark = (
+        SparkSession.builder
+        .config(conf=spark_conf)
+        .config("spark.sql.broadcastTimeout", "36000")
+        .master('local[*]')
+        .getOrCreate()
+    )
 
-    TEP_raw_data = []
+    spark.sparkContext.addFile("https://www.thesgc.org/sites/default/files/fileuploads/available-teps.tsv")
 
-    for row in TEP_table.find('tbody').findAll('tr'):
-        TEP_raw_data.append({
-            'targetFromSourceId': get_gene_name(row),
-            'url': get_url(row),
-            'therapeuticArea': get_therapeutic_area(row),
-            'description': get_description(row)
-        })
+    # Fetching and processing the TEP table and saved as a JSON file:
+    TEP_df = (
+        spark.read.csv(SparkFiles.get("available-teps.tsv"), sep='\t', header=True)
 
-    return pd.DataFrame(TEP_raw_data).explode('targetFromSourceId')
+        # Generating TEP url from Gene column: SLC12A4/SLC12A6 -> https://www.thesgc.org/tep/SLC12A4SLC12A6
+        .withColumn('url', concat(lit('https://www.thesgc.org/tep/'), regexp_replace(lower(col('Gene')), '/', '')))
 
+        # Exploding TEPs, where multiple genes are given:
+        .withColumn('targetFromSourceId', explode(split(col('Gene'), '/')))
 
-def main(outputFile: str) -> None:
+        # Renaming columns:
+        .withColumnRenamed('Therapeutic Area', 'therapeuticArea')
+        .withColumnRenamed('Description', 'description')
 
-    # The TEP list compiled as dataframe:
-    tep_list = retrieve_tep_table()
-    logging.info(f'Number of TEPs retrieved: {len(tep_list)}')
-    logging.info(f'Number of unique gene symbols in the tep: {len(tep_list.targetFromSourceId.unique())}')
+        # Dropping columns:
+        .drop(*['Gene', 'version', 'Date'])
+        .persist()
+    )
 
-    if tep_list.targetFromSourceId.duplicated().any():
-        duplicated_targets = tep_list.loc[tep_list.targetFromSourceId.duplicated()]
-        logging.error(f'The following symbols were not unique: {duplicated_targets}')
-        raise ValueError('The target symbol list is expected to be unique! Check logs for deatils.')
+    logging.info('TEP dataset has been downloaded and formatted.')
+    logging.info(f'Number of TEPs: {TEP_df.count()}')
+    logging.info(f'Number of unique genes: {TEP_df.select("targetFromSourceId").distinct().count()}')
 
     # Saving data:
-    logging.info(f'Saving data to {outputFile}.')
-    tep_list.to_json(outputFile, compression='infer', orient='records', lines=True)
+    write_evidence_strings(TEP_df, outputFile)
```

Comment:
Since we are no longer talking of evidence strings, I'd suggest to change the name of this function to something like `write_json_strings`/`write_df_to_json`/`export_to_json`. I'd also change the name of `common.evidence.py` to `common.utils.py`
```suggestion
    write_json_strings(TEP_df, outputFile)
```

Reply:
I 100% with this call, however let's merge this version as it is, and have an other PR with a ticket so this update would be more transparent.